crates/revm/src/instructions/bitwise.rs

```diff
@@ -48,41 +48,41 @@ pub fn byte(op1: U256, op2: U256) -> U256 {
 }
 
 pub fn shl(shift: U256, value: U256) -> U256 {
-    if value.is_zero() || shift >= U256::from(256) {
-        U256::zero()
+    if value == U256::ZERO || shift >= U256::from(256) {
+        U256::ZERO
     } else {
-        let shift: u64 = shift.as_u64();
+        let shift = u64::try_from(shift).unwrap();
         value << shift as usize
     }
```

Comment:
Unlike `std`, the shifts in ruint don't panic when the argument is large, so `value << shift.try_into::<usize>().ok_or(256)` should work.